lib/src/gotrue_client.dart

```diff
@@ -216,6 +217,51 @@ class GoTrueClient {
         redirectTo: redirectTo, scopes: scopes, queryParams: queryParams);
   }
 
+  /// Allows signing in with an ID token issued by certain supported providers.
+  /// The [idToken] is verified for validity and a new session is established.
+  @Experimental()
+  Future<AuthResponse> signInWithIdToken({
+    required Provider provider,
+    required String idToken,
+    String? nonce,
+    String? captchaToken,
+  }) async {
+    _removeSession();
+
+    if (provider != Provider.google && provider != Provider.apple) {
```

Comment:
I believe this must be in documentation, because Github and Facebook sign in is pretty common

Reply:
Like in the comments about the method? Inside the `///` ?